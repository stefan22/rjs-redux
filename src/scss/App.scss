@import './base/base';
@import './base/settings';
// @import './components/expenses-list';
// @import './components/top-bar';





/* topbar */

.topbar {
  flex-grow: 1;
}
.topbar-icon {
  margin-left: -12;
  margin-right: 20;
}
.topbar-grow {
  flex-grow: 1;
}
.topbar-icon img {//doc logo
  max-width: 125px;
  width:100%;
  height: auto;
}

.topbar-home:hover,.topbar-home:active, .topbar-add:hover,.topbar-add:active, .topbar-create:hover, .topbar-create:active, .topbar-edit:hover, .topbar-edit:active, .topbar-delete:active, .topbar-help:active,
.topbar-delete:hover, .topbar-help:hover, .back-home:active, .back-home:focus,
.back-home:hover, a.button:hover, .button:hover, button:hover  {
  border:none;
}

a.nav.topbar-home, a.nav.topbar-add, a.nav.topbar-help  {
  font-size: $nav-size;
  color:#d2d2d2;
  padding: 18px 20px;
  text-decoration: none;
  font-weight: 300;
  text-transform: capitalize;
  &.is-active {
    color:#fff;
  }

}

button.topbar-icon:hover, button.topbar-icon:active {
  border:none;
}

/* button submenu */

.button-wrap button {
  font-size: $reg-button-size;
  padding: 15px 27px;
}


/* lost found page*/
.lost-header {
  margin:2em 0;
  h1,h4 {
    text-align: center;
  }
}
.lost-error {
  display: block;
  margin: 0 auto;
  max-width: 600px;
  width: 100%;
  a {
    overflow: auto;
    &.back-home {
      padding: 0;
      background-color: transparent;
      &:hover {
        background-color:transparent !important;
      }

      img {
        height: auto;
        max-width: 600px;
        width:100%;
        border-radius: 4px;
      }

    }

  }


}
