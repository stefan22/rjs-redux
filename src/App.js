import React, { Component } from 'react';
import Header from './components/header';
import BlueButton from './components/button';
import './scss/App.scss';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      pageName: 'Home page'
    }
  }
  render() {
    let btnName = 'Page Not Found';
    return (
      <div className="container">
          <Header pageName={this.state.pageName} />

          <div className='main-content'>
            <h2>Main content</h2>
            <p className='text-left'>Dolor primis viverra aenean fusce inceptos bibendum sagittis pretium aptent inceptos nulla facilisi. Laoreet molestie est faucibus dapibus in sed aptent quisque nulla. Lacinia, taciti vitae orci ante tincidunt in. Erat ac; convallis dignissim? Sollicitudin senectus, interdum imperdiet. Tortor urna gravida nisi libero neque tristique mauris inceptos amet ligula. Vulputate sociis eget at etiam inceptos quisque eleifend.</p>

            <BlueButton name={btnName} />

            <h2>Heading</h2>
            <p className='text-left'>Consequat lorem sollicitudin fames suscipit, tristique tempor egestas tortor nisl. Mollis hendrerit consectetur facilisis tincidunt ipsum ut ultricies pulvinar est convallis. Augue quisque varius praesent. Convallis lacinia nunc et cursus sit sagittis pellentesque purus viverra. Ridiculus ligula fusce natoque per neque metus natoque habitasse tincidunt tempus quam potenti! Nostra euismod tortor venenatis lorem laoreet lacinia ante congue potenti enim sollicitudin. Tincidunt, ac taciti felis et curabitur phasellus vestibulum commodo dictum adipiscing cursus. Netus vehicula eros.</p>
            <p className='text-left'>
            Dolor primis viverra aenean fusce inceptos bibendum sagittis pretium aptent inceptos nulla facilisi. Laoreet molestie est faucibus dapibus in sed aptent quisque nulla. Lacinia, taciti vitae orci ante tincidunt in. Erat ac; convallis dignissim? Sollicitudin senectus, interdum imperdiet. Tortor urna gravida nisi libero neque tristique mauris inceptos amet ligula. Vulputate sociis eget at etiam inceptos quisque eleifend.
            </p>
          </div>
      </div>
    );
  }
}




export default App;









