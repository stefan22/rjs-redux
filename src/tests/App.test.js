import React from 'react';
import {shallow, mount, render} from 'enzyme';
import toJSON from 'enzyme-to-json';
import App from '../App';

describe('App component', () => {
  it('should render home page component correctly', () => {
    const wrapper = shallow(<App />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should contain an h2 tag of `Main content`', () => {
    const wrapper = shallow(<App />);
    const main = <h2>Main content</h2>;
    expect(wrapper.containsMatchingElement(main)).toEqual(true);
  });


});

