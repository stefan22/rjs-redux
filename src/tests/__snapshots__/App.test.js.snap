// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render home page component correctly 1`] = `
<div
  className="container"
>
  <Header
    pageName="Home page"
    subtitle="home page dashboard"
  />
  <div
    className="main-content"
  >
    <h2>
      Main content
    </h2>
    <p
      className="text-left"
    >
      Dolor primis viverra aenean fusce inceptos bibendum sagittis pretium aptent inceptos nulla facilisi. Laoreet molestie est faucibus dapibus in sed aptent quisque nulla. Lacinia, taciti vitae orci ante tincidunt in. Erat ac; convallis dignissim? Sollicitudin senectus, interdum imperdiet. Tortor urna gravida nisi libero neque tristique mauris inceptos amet ligula. Vulputate sociis eget at etiam inceptos quisque eleifend.
    </p>
    <h2>
      Heading
    </h2>
    <p
      className="text-left"
    >
      Consequat lorem sollicitudin fames suscipit, tristique tempor egestas tortor nisl. Mollis hendrerit consectetur facilisis tincidunt ipsum ut ultricies pulvinar est convallis. Augue quisque varius praesent. Convallis lacinia nunc et cursus sit sagittis pellentesque purus viverra. Ridiculus ligula fusce natoque per neque metus natoque habitasse tincidunt tempus quam potenti! Nostra euismod tortor venenatis lorem laoreet lacinia ante congue potenti enim sollicitudin. Tincidunt, ac taciti felis et curabitur phasellus vestibulum commodo dictum adipiscing cursus. Netus vehicula eros.
    </p>
    <p
      className="text-left"
    >
      Dolor primis viverra aenean fusce inceptos bibendum sagittis pretium aptent inceptos nulla facilisi. Laoreet molestie est faucibus dapibus in sed aptent quisque nulla. Lacinia, taciti vitae orci ante tincidunt in. Erat ac; convallis dignissim? Sollicitudin senectus, interdum imperdiet. Tortor urna gravida nisi libero neque tristique mauris inceptos amet ligula. Vulputate sociis eget at etiam inceptos quisque eleifend.
    </p>
  </div>
</div>
`;
